src/contexts/TokenPriceContext.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {
+  useEffect,
+  useContext,
+  createContext,
+  useRef,
+  useState,
+} from "react"
+import { exchangeAPI, CoingeckoID } from "../utils"
+
+export const TokenPriceContext = createContext<{
+  threshold: string
+}>({
+  threshold: "0",
+})
+
+export const TokenPriceContextProvider: React.FC = ({ children }) => {
+  const shouldUpdateState = useRef<boolean>(true)
+  const [thresholdPrice, setThresholdPrice] = useState("0")
+
+  useEffect(() => {
+    shouldUpdateState.current = true
+
+    const fetchTokenPrice = async () => {
+      const tPrice = await exchangeAPI.fetchCryptoCurrencyPriceUSD(
+        CoingeckoID.T
+      )
+
+      if (!shouldUpdateState.current) return
+      setThresholdPrice(tPrice)
+    }
+
+    fetchTokenPrice()
+
+    return () => {
+      shouldUpdateState.current = false
+    }
+  }, [])
```

Comment:
I think we're missing some dependencies in this array like `shouldUpdateState`. If the intent is to only run this effect one time we could probably get rid of that boolean. What do you think?

Reply:
`shouldUpdateState` is a ref so there is no need to include it in the hook dependency. The idea behind `shouldUpdateState` flag is to avoid updating the component state when it is actually unmounted.